src/Service.Host/client/src/components/PurchaseOrders/PurchaseOrderUtility.js

```diff
@@ -0,0 +1,767 @@
+import Tooltip from '@mui/material/Tooltip';
+import Close from '@mui/icons-material/Close';
+import PrintIcon from '@mui/icons-material/Print';
+import Email from '@mui/icons-material/Email';
+import Send from '@mui/icons-material/Send';
+import Alert from '@mui/material/Alert';
+import AlertTitle from '@mui/material/AlertTitle';
+import Collapse from '@mui/material/Collapse';
+import { makeStyles } from '@mui/styles';
+import {
+    Page,
+    SaveBackCancelButtons,
+    collectionSelectorHelpers,
+    Typeahead,
+    InputField,
+    SnackbarMessage,
+    itemSelectorHelpers,
+    Loading,
+    Dropdown,
+    TypeaheadTable,
+    userSelectors,
+    getItemError,
+    ErrorCard,
+    processSelectorHelpers
+} from '@linn-it/linn-form-components-library';
+import currenciesActions from '../../actions/currenciesActions';
+import employeesActions from '../../actions/employeesActions';
+import nominalsActions from '../../actions/nominalsActions';
+import countriesActions from '../../actions/countriesActions';
+import suppliersActions from '../../actions/suppliersActions';
+import partsActions from '../../actions/partsActions';
+import history from '../../history';
+import config from '../../config';
+import purchaseOrderActions from '../../actions/purchaseOrderActions';
+
+function PurchaseOrderUtility({ creating }) {
+    const dispatch = useDispatch();
+    const clearErrors = () => dispatch(purchaseOrderActions.clearErrorsForItem());
+
+    const { orderNumber } = useParams();
+
+    useEffect(() => {
+        if (orderNumber) {
+            dispatch(purchaseOrderActions.fetch(orderNumber));
+        } else if (creating) {
+            dispatch(purchaseOrderActions.fetchState());
+        }
+    }, [orderNumber, dispatch, creating]);
+
+    const item = useSelector(reduxState => itemSelectorHelpers.getItem(reduxState.purchaseOrder));
+    const loading = useSelector(state =>
+        creating
+            ? itemSelectorHelpers.getApplicationStateLoading(state.purchaseOrderApplicationState)
+            : itemSelectorHelpers.getItemLoading(state.purchaseOrder)
+    );
+
+    const itemError = useSelector(state => getItemError(state, 'purchaseOrder'));
+    const [order, setOrder] = useState({});
+
+    useEffect(() => {
+        if (item?.orderNumber) {
+            setOrder(item);
+        } else {
+            dispatch(purchaseOrderActions.clearErrorsForItem());
+        }
+    }, [item, dispatch]);
+
+    const suppliersSearchResults = useSelector(state =>
+        collectionSelectorHelpers.getSearchItems(state.suppliers, 100, 'id', 'id', 'name')
+    );
+    const suppliersSearchLoading = useSelector(state =>
+        collectionSelectorHelpers.getSearchLoading(state.suppliers)
+    );
+    const searchSuppliers = searchTerm => dispatch(suppliersActions.search(searchTerm));
+
+    const partsSearchResults = useSelector(state =>
+        collectionSelectorHelpers.getSearchItems(state.parts)
+    ).map?.(c => ({
```

Comment:
I would hope getSearchItems only ever turns an array. If thats true then you can ditch this ? mark probably